tests/jupyterhub_api/test_create_session.py

```diff
@@ -0,0 +1,151 @@
+    parser.add_argument("--port", dest="port", type=int,
+                        required = True,
+                        )
+
+    parser.add_argument( "--users", nargs='+', dest="users",
+                        required  = True,
+                        help='list of users')
+    parser.add_argument("--delay", dest="delay",
+                        required=True,
+                        help='delay while creating the server')
+
+    parser.add_argument("--json", dest="json",
+                         required=True,
+                         help = 'json data')
+
+    parser.add_argument("--token", dest = "token",
+                        required= False,
+                        help='token')
+
+    parser.add_argument("--base_path", dest="base",
+                        required=False,
+                        help='base path')
+    args = parser.parse_args()
+
+    return args
+
+
+class CreateSession(JupyterhubTest):
+    """
+    Create a session
+
+    To run the test run the following command:
+
+     python3 test_create_session.py --port 443 --users user3 --delay 30 --json '{"LCG-rel": "LCG_95a", "platform": "x86_64-centos7-gcc7-opt", "scriptenv": "none", "ncores": 2, "memory": 8589934592, "spark-cluster": "none"}'
+
+
+    For multiple users
+
+    python3 test_create_session.py --port 443 --users user0 user1 user2 --delay 30 --json '{"LCG-rel": "LCG_95a", "platform": "x86_64-centos7-gcc7-opt", "scriptenv": "none", "ncores": 2, "memory": 8589934592, "spark-cluster": "none"}'
+
+    """
+    def __init__(self, port, token, users, data, delay,  base_path, verify):
+        param = {}
+        param['test_name'] = 'Session_Creation_test'
+        JupyterhubTest.__init__(self, port,  token, base_path, verify, **param)
+        self.data = data
+        self.users = users
+        self.delay = delay
+
+
+    def check_create_session(self):
+
+        self.log.write("info", "creating servers..")
+        r = ""
+        for user in self.users:
+            try:
+                r = super().api_calls("post", user, data = self.data, endpoint ="/server")
+            except requests.exceptions.RequestException as e:
+                self.exit = 1
+                self.log.write("error", "status code " + str(r.status_code) + " " + str(e))
+
+            else:
+                if r.status_code == 202:
+                    self.log.write("info", "Successfully requested the server")
+                    """
+                     After requesting a server it needs to respond within 30 seconds otherwise there will be 
+                     Timeout Error
+                     Example:
+        
+                     TimeoutError: Server at http://172.18.0.15:8888/user/user0/ didn't respond in 30 seconds
+                     So, even after a successful request a server may not be created
+                    """
+                    """
```

Comment:
Single line comment, use #

Reply:
Outdated file, the changes have been merged.